File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetPhysicalDeviceImageFormatProperties.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkGetPhysicalDeviceImageFormatProperties_callNative(JNIEnv* env, jclass jc, jlong address, jlong physicalDevice, jint format, jint type, jint tiling, jint usage, jint flags, jlong pImageFormatProperties){
    (void)env;
    (void)jc;
    PFN_vkGetPhysicalDeviceImageFormatProperties f = (PFN_vkGetPhysicalDeviceImageFormatProperties)jniLongToFunctionPointer(address);
    return f(
        *((VkPhysicalDevice*)jniLongToPointer(physicalDevice)),
        (VkFormat)format,
        (VkImageType)type,
        (VkImageTiling)tiling,
        (VkImageUsageFlags)usage,
        (VkImageCreateFlags)flags,
        ((VkImageFormatProperties*)jniLongToPointer(pImageFormatProperties))
    );
}

File: 1.1/src/cz/mg/vulkan/PFNvkGetPhysicalDeviceImageFormatProperties.java
package cz.mg.vulkan;

public class PFNvkGetPhysicalDeviceImageFormatProperties extends VkFunctionPointer {
    public PFNvkGetPhysicalDeviceImageFormatProperties() {
    }

    protected PFNvkGetPhysicalDeviceImageFormatProperties(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetPhysicalDeviceImageFormatProperties(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetPhysicalDeviceImageFormatProperties(long value) {
        setValue(value);
    }

    public PFNvkGetPhysicalDeviceImageFormatProperties(VkInstance instance) {
        super(instance, new VkString("vkGetPhysicalDeviceImageFormatProperties"));
    }

    public int call(VkPhysicalDevice physicalDevice, int format, int type, int tiling, int usage, int flags, VkImageFormatProperties pImageFormatProperties){
        return callNative(getValue(), physicalDevice != null ? physicalDevice.getVkAddress() : VkPointer.getNullAddressNative(), format, type, tiling, usage, flags, pImageFormatProperties != null ? pImageFormatProperties.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long physicalDevice, int format, int type, int tiling, int usage, int flags, long pImageFormatProperties);

}
